gm4_double_doors/data/gm4_double_doors/functions/main.mcfunction

```diff
@@ -0,0 +1,4 @@
+execute as @a[scores={gm4_double_doors_auto_toggle_settings_trigger=1..}] run function gm4_double_doors:toggle_auto_setting
+scoreboard players enable @a gm4_double_doors_auto_toggle_settings_trigger
+
+schedule function gm4_double_doors:main 20t
```

Comment:
Is any of this really necessary?
We could simply check the triggered score itself (`gm4_double_doors_auto_toggle_settings_trigger` side note: a shorter name might be nice) in the `.../toggle.mcfunction` functions and allow any non-zero positive score to inhibit the auto-closing feature (`unless score @s gm4_double_doors_auto_toggle_settings_trigger matches 1..`).

That way almost no code in `main` is necessary. Obviously, we still have to enable the trigger for everyone (and keep enabling it once they have used it... annoyingly), but maybe we can find a way of doing that without a `main` even? (to be researched)